GitUI/UserControls/RevisionGrid/Columns/AvatarColumnProvider.cs

```diff
@@ -13,15 +13,15 @@ namespace GitUI.UserControls.RevisionGrid.Columns
     internal sealed class AvatarColumnProvider : ColumnProvider
     {
         private readonly RevisionDataGridView _revisionGridView;
-        private readonly IAvatarProvider _avatarProvider;
+        private readonly ICachedAvatarProvider _avatarProvider;
 
-        public AvatarColumnProvider(RevisionDataGridView revisionGridView, IAvatarProvider avatarProvider)
+        public AvatarColumnProvider(RevisionDataGridView revisionGridView, ICachedAvatarProvider avatarProvider)
             : base("Avatar")
         {
             _revisionGridView = revisionGridView;
             _avatarProvider = avatarProvider;
 
-            _avatarProvider.CacheCleared += _revisionGridView.Invalidate;
+            _avatarProvider.CacheCleared += (sender, args) => _revisionGridView.Invalidate();
```

Comment:
RevisionGridControl refreshed after closing the settings window.

@RussKie, is there a need to update the grid before closing the settings window?

Reply:
If avatar settings change and avatar cache is invalidated - we need to refresh the grid to download new avatars, and show them.

Reply:
Avatar cache can be invalidated from the commit info panel
![image](https://user-images.githubusercontent.com/4403806/99904687-9b248f80-2d20-11eb-8902-837812981a9c.png)

Reply:
I suggest to transfer this to the service level (AvatarService).

Reply:
We can do it as a follow up. This PR has grown large as is.